praw/exceptions.py

```diff
@@ -185,6 +276,38 @@ def __init__(self, url: str, message: str = "Invalid URL: {}"):
         super().__init__(message.format(url))
 
 
+class MediaPostFailed(RedditAPIException):
+    """Indicate exceptions where invalid media was uploaded."""
+
+    @property
+    def error_type(self) -> str:
+        """Constant representing the error type of all instances."""
+        return "Failed Media Upload"
+
+    @property
+    def message(self) -> str:
+        """Constant representing the message of all instances."""
+        return (
+            "The attempted media upload action has failed. Possible causes"
+            " include the corruption of media files. Check that the media "
+            "file can be opened on your local machine."
+        )
+
+    @property
```

Comment:
It seems like these properties could simply be defined as instance variables inside `__init__`. Do we need properties for them?

Reply:
I could define them as instance variables, I just wanted to override the deprecated variables.

Reply:
Overriding in this way is correct. Otherwise you’d have to make the properties writable on the parent class.

I’d would also like to point out that you can’t override properties with fields in C# (a strongly typed language).

Reply:
Good point.